internal/characters/tartaglia/skill.go

```diff
@@ -9,29 +9,80 @@ import (
 	"github.com/genshinsim/gcsim/pkg/core/glog"
 )
 
-var skillMeleeFrames []int
-var skillRangedFrames []int
+var (
+	skillMeleeFrames      []int
+	skillMeleeWalkFrames  []int
+	skillMeleeDashFrames  []int
+	skillRangedFrames     []int
+	skillRangedWalkFrames []int
+	skillRangedDashFrames []int
+)
 
-const skillHitmark = 28
+const (
+	skillHitmark     = 16
+	skillWalkHitmark = 3
+	skillDashHitmark = 3
+)
 
 func init() {
 	// skill (melee) -> x
-	skillMeleeFrames = frames.InitAbilSlice(20)
-
+	skillMeleeFrames = frames.InitAbilSlice(18)
+	skillMeleeFrames[action.ActionAttack] = 17
+	skillMeleeFrames[action.ActionBurst] = 18
+	skillMeleeFrames[action.ActionDash] = 17
+	skillMeleeFrames[action.ActionJump] = 17
+	skillMeleeFrames[action.ActionSwap] = 16
+	skillMeleeWalkFrames = frames.InitAbilSlice(24)
```

Comment:
walk -> e frames should either be extended by the 4-1f (walk takes 1f) that you walk beforehand or you overwrite the walk action to be 4f.